src/utils/messaging-outbox/src/agent/outbox_agent.rs

```diff
@@ -166,6 +166,16 @@ impl OutboxAgent {
             .try_collect::<Vec<_>>()
             .await?;
 
+        let latest_message_ids_by_producer = {
```

Comment:
Minor: add a comment why we are doing this. This is quite a complex part of the system, and you can see a lot of comments around.